lib/ansible/modules/cloud/digital_ocean/do_sshkeys_facts.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2015, Patrick F. Marques <patrickfmarques@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: do_sshkeys_facts
+short_description: DigitalOcean SSH keys facts
+description:
+     - Fetch DigitalOcean SSH keys facts.
+version_added: "2.4"
+author: "Patrick Marques (@pmarques)"
+options:
+  oauth_token:
+    description:
+     - DigitalOcean API token.
+    required: true
+
+notes:
+  - Version 2 of DigitalOcean API is used.
+requirements:
+  - "python >= 2.6"
+'''
+
+
+EXAMPLES = '''
+- name: "List all sshkeys"
+  do_sshkeys_facts:
+    register: result
+'''
```

Comment:
This module was born as part of the do_sshkeys and end up using this to check keys in a cluster. So, for my use case it is just

```
- debug: var= result
```

The **RETURN** explains the data structure of *result* and so, this can be used as any other variable to test / conditional run tasks if someone needs.